core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerAPI.java

```diff
@@ -364,6 +404,57 @@ public interface RegisteredTable {
         void onTable(int tableId, final CharSequence tableName, long lastTxn);
     }
 
+    /**
+     * Reads table id and last txn from both sequencer metadata and table transaction log.
+     * Used to avoid creating a TableSequencer in {@link #forAllWalTables}.
+     */
+    private static class LastWalTxnReader implements Closeable {
+        private final FilesFacade ff;
+        private final MemoryMR roMetaMem;
+        private final MemoryMR roTxnMem;
+        private long maxTxn;
+        private int tableId;
+
+        public LastWalTxnReader(FilesFacade ff) {
+            this.ff = ff;
+            roMetaMem = Vm.getMRInstance();
+            roTxnMem = Vm.getMRInstance();
+        }
+
+        @Override
+        public void close() {
+            reset();
+            Misc.free(roMetaMem);
+            Misc.free(roTxnMem);
+        }
+
+        public int getTableId() {
+            return tableId;
+        }
+
+        public void open(Path path, int pathLen) {
+            reset();
+            openSmallFile(ff, path, pathLen, roMetaMem, META_FILE_NAME, MemoryTag.MMAP_SEQUENCER_METADATA);
```

Comment:
It would be lighter to use `ff.readULong()` to read 1 value from a file instead of mapping it

Reply:
Good idea. Will change the code.

Reply:
Added `ff.readInt()` method and renamed `ff.readULong()` to `readLong` as it doesn't return an unsigned long. See e619cb265c3127620efd8e8b5e14c4e004ef6460

Will do a follow-up commit with the binary files.

Reply:
Negative result of `readULong` means error.

Reply:
You can read `Long` instead of `Int`, files should be long enough to fead 4 extra bytes so you can avoid adding new native methods

Reply:
Then I would need to deal with endianness. Adding `readInt` method is much more straightforward.

Reply:
> Negative result of `readULong` means error.

Then it's still not a real unsigned 64-bit integer, but rather an unsigned 63-bit integer.

Reply:
Renamed `FilesFacade` methods to `readNonNegativeInt` and `readNonNegativeLong` to make them less error-prone.